engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/callcaching/EngineJobHashingActor.scala

```diff
@@ -1,282 +1,126 @@
-    fileContentHashesNeeded.foreach(fileHashingActor ! _)
-  }
-
-  private def calculateInitialHashes(nonFileInputs: Iterable[WdlValueSimpleton], fileInputs: Iterable[WdlValueSimpleton]): Set[HashResult] = {
-
-    val commandTemplateHash = HashResult(HashKey("command template"), jobDescriptor.call.task.commandTemplateString.md5HashValue)
-    val backendNameHash = HashResult(HashKey("backend name"), backendName.md5HashValue)
-    val inputCountHash = HashResult(HashKey("input count"), (nonFileInputs.size + fileInputs.size).toString.md5HashValue)
-    val outputCountHash = HashResult(HashKey("output count"), jobDescriptor.call.task.outputs.size.toString.md5HashValue)
-
-    val runtimeAttributeHashes = runtimeAttributeDefinitions map { definition => jobDescriptor.runtimeAttributes.get(definition.name) match {
-      case Some(wdlValue) => HashResult(HashKey("runtime attribute: " + definition.name, definition.usedInCallCaching), wdlValue.valueString.md5HashValue)
-      case None => HashResult(HashKey("runtime attribute: " + definition.name, definition.usedInCallCaching), UnspecifiedRuntimeAttributeHashValue)
-    }}
-
-    val inputHashResults = nonFileInputs map {
-      case WdlValueSimpleton(name, value) => HashResult(HashKey(s"input: ${value.wdlType.toWdlString} $name"),  value.toWdlString.md5HashValue)
+                            jobDescriptor: BackendJobDescriptor,
+                            initializationData: Option[BackendInitializationData],
+                            fileHashingActorProps: Props,
+                            callCacheReadingJobActorProps: Props,
+                            runtimeAttributeDefinitions: Set[RuntimeAttributeDefinition],
+                            backendName: String,
+                            activity: CallCachingActivity) extends Actor with ActorLogging with JobLogging {
+
+  override val jobTag = jobDescriptor.key.tag
+  override val workflowId = jobDescriptor.workflowDescriptor.id
+
+  private [callcaching] var initialHash: Option[InitialHashingResult] = None
+
+  private [callcaching] val callCacheReadingJobActor = if (activity.readFromCache) {
+    Option(context.actorOf(callCacheReadingJobActorProps))
+  } else None
+
+  override def preStart(): Unit = {
+    context.actorOf(CallCacheHashingJobActor.props(
+      jobDescriptor,
+      callCacheReadingJobActor,
+      initializationData,
+      runtimeAttributeDefinitions,
+      backendName,
+      fileHashingActorProps,
+      activity.writeToCache
+    ))
+    super.preStart()
+  }
+
+  override def receive = {
+    case initialHashResult: InitialHashingResult => initialHash = Option(initialHashResult)
+    case finalFileHashResult: FinalFileHashingResult => sendHashes(finalFileHashResult)
+    case CacheMiss => receiver ! CacheMiss
+    case hit: CacheHit => receiver ! hit
+    case NextHit =>
+      callCacheReadingJobActor match {
+        case Some(readActor) =>
+          readActor ! NextHit
```

Comment:
I don't think so ? The `callCacheReadingJobActor` only shuts down if it determines it's a `CacheMiss` at which point the EJEA will stop asking for `NextHit`